### DISCUSSION TITLE
pyinstaller created .exe can't find module ortools.constraint_solver.routing_parameters_pb2

### DISCUSSION THREAD

**Discussion by QuentinBennett** [google/or-tools#3579] [Build and configuration issues]
**What version of OR-Tools and what language are you using?**
Version: 9.4.1874
Language: Python 3.10

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
Routing

**What operating system (Linux, Windows, ...) and version?**
Windows Server 2016 Datacenter

**What did you do?**
Script works fine under python.
Package created using
pyinstaller --hidden-import ortools.constraint_solver.routing_parameters_pb2 --hidden-import pkg_resources.py2_warn --onefile .\vrptime.py

(Hidden imports suggested in https://github.com/google/or-tools/issues/802 )

**What did you expect to see**
Program to run as per script

**What did you see instead?**
Traceback (most recent call last):
  File "vrptime.py", line 211, in <module>
    main(driverId)
  File "vrptime.py", line 193, in main
    search_parameters = pywrapcp.DefaultRoutingSearchParameters()
  File "ortools\constraint_solver\pywrapcp.py", line 4680, in DefaultRoutingSearchParameters
ModuleNotFoundError: No module named 'ortools.constraint_solver.routing_parameters_pb2'
[6736] Failed to execute script 'vrptime' due to unhandled exception!

Make sure you include information that can help us debug (full error message, model Proto).

**Anything else we should know about your project / environment**
VC_redist.x64.exe installed

**Comment by Mizux** [comment]
`-p DIR, --paths DIR`
A path to search for imports (like using PYTHONPATH). Multiple paths are allowed, separated by ':', or use this option multiple times. Equivalent to supplying the pathex argument in the spec file.

`--hidden-import MODULENAME, --hiddenimport MODULENAME`
Name an import not visible in the code of the script(s). This option can be used multiple times.

`--collect-submodules MODULENAME`
Collect all submodules from the specified package or module. This option can be used multiple times.

`--collect-data MODULENAME, --collect-datas MODULENAME`
Collect all data from the specified package or module. This option can be used multiple times.

`--collect-binaries MODULENAME`
Collect all binaries from the specified package or module. This option can be used multiple times.

`--collect-all MODULENAME`
Collect all submodules, data files, and binaries from the specified package or module. This option can be used multiple times.

src: https://pyinstaller.org/en/stable/usage.html?highlight=hidden#cmdoption-hidden-import

**Reply by QuentinBennett** [reply]
Thanks for the answer, however, it doesn't help much as I wouldn't know which options or values to use.

Given that python and all modules have been installed using the default settings, shouldn't everything be in the expected place?

**Reply by lperron** [reply]
pyinstaller is not very bright :-(
